src/distribution/robust_soliton.rs

```diff
@@ -0,0 +1,345 @@
+use crate::distribution::{Soliton, IdealSoliton};
+use crate::statistics::*;
+use crate::{Result, StatsError};
+use rand::distributions::Distribution;
+use rand::Rng;
+use std::f64;
+
+/// Implements the [Robust Soliton]
+/// (https://en.wikipedia.org/wiki/Soliton_distribution)
+/// distribution
+///
+/// # Examples
+///
+/// ```
+/// use statrs::distribution::{Soliton, IdealSoliton};
+/// let n = RobustSoliton::new(10, true, 0.1, 0.1).unwrap();
+/// assert_eq!(n.soliton(5), 0.05879983550709325);
+/// ```
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct RobustSoliton {
+    min: i64,
+    max: i64,
+    spike: Option<i64>,
+    cumulative_probability_table: Vec<f64>,
+    ripple: Ripple,
+    fail_probability: f64,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct RobustSolitonDistribution {
+    ripple: Ripple,
+    fail_probability: f64,
+}
+
+impl RobustSoliton {
+    /// Constructs a new robust soliton distribution with a maximum value of `max`.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if `max < 0`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use statrs::distribution::RobustSoliton;
+    ///
+    /// let mut result = RobustSoliton::new(10, true, 0.1, 0.1);
+    /// assert!(result.is_ok());
+    ///
+    /// result = RobustSoliton::new(-1, true, 0.1, 0.1);
+    /// assert!(result.is_err());
+    /// ```
+    pub fn new(max: i64, heuristic: bool, ripple: f64, fail_probability: f64) -> Result<RobustSoliton> {
+        if max < 1 {
+            Err(StatsError::BadParams)
+        } else {
+            let pmf_table: Vec<f64> = Vec::with_capacity(max as usize);
```

Comment:
This type is inferred and can be elided.

Reply:
:+1: